### ISSUE TITLE
Support semicolon (;) as CSV separator

### ISSUE THREAD

**Issue by DamianFekete** [mikefarah/yq#1950]
**Please describe your feature request.**
I wish I could use yq to parse CSV files where fields are separated by semicolons instead of commas.

**Describe the solution you'd like**
If we have input.csv like:
```csv
Header A;Header B
A1;B1
A2;B2
```

And we run a command:

```bash
yq -p csv -o json --csv-separator ; input.csv
# autodetection (maybe)
yq -p csv -o json input.csv
```

it could output

```yaml
[
  {
    "Header A": "A1",
    "Header B": "B1"
  },
  {
    "Header A": "A2",
    "Header B": "B2"
  }
]
```

**Describe alternatives you've considered**
Using another tool

**Additional context**
Different locales have a different field separator than comma, as comma may be used for numbers instead of the English decimal point.

**Comment by mikefarah** [comment]
I like it, will be available in the next release (no auto-detection though sorry)

**Comment by mikefarah** [comment]
Fixed in 4.42.1
